compiler/rustc_infer/src/infer/higher_ranked/mod.rs

```diff
@@ -24,9 +24,9 @@ impl<'a, 'tcx> CombineFields<'a, 'tcx> {
         // as-is, we need to do some extra work here in order to make sure
         // that function subtyping works correctly with respect to regions
         //
-        // Note: this is a subtle algorithm.  For a full explanation,
-        // please see the large comment at the end of the file in the (inlined) module
-        // `doc`.
+        // Note: this is a subtle algorithm.  For a full explanation, please see
+        // the rustc book:
```

Comment:
nit: can you squash https://github.com/rust-lang/rust/pull/78177/commits/9d49d63300bbea5fbec89e9e36be4b88ddb3336d into https://github.com/rust-lang/rust/pull/78177/commits/185ce4636c4f194b8ecb22e5e16e1c38004067e7 ?